### Fuel Hazard Tahoe

In [3]:
!ls Historical*

HistoricalFuelTreatments.zip

HistoricalFuelTreatments:
USFS_Hazard_Fuel_Teatmentt_Activity_webmerc.dbf
USFS_Hazard_Fuel_Teatmentt_Activity_webmerc.prj
USFS_Hazard_Fuel_Teatmentt_Activity_webmerc.qpj
USFS_Hazard_Fuel_Teatmentt_Activity_webmerc.shp
USFS_Hazard_Fuel_Teatmentt_Activity_webmerc.shx

HistoricalFuelTreatments.zip.download:


In [4]:
import geopandas as gpd

In [75]:
dateparse = lambda dates: [pd.datetime.strptime(d, '%Y-%m-%d') for d in dates]
infile = 'HistoricalFuelTreatments/USFS_Hazard_Fuel_Teatmentt_Activity_webmerc.shp'
treatments = gpd.read_file(infile) #, parse_dates=['REV_DATE'], date_parser=dateparse)

In [76]:
treatments.head(1)

SUID     ORG ACTIVITY_C                          ACTIVITY  \
0  0519540030061003000  051954       1153  Piling of Fuels, Hand or Machine   

  LOCAL_QUAL  ASU_NBR_UN ASU_UOM ADMIN_REGI ADMIN_FORE ADMIN_DIST  \
0        N/A         4.0   ACRES         05         19         54   

                         ...                         ETL_MODIFI  \
0                        ...                                  N   

                UK  EDW_INSERT                             UK_HAZ  \
0  0519_12777_ACTP  2017-01-14  5215581010602_0519540030061003000   

            CRC_HAZ  ETL_MODI_1  GIS_ACRES  SHAPE_AREA SHAPE_LEN  \
0  B1AFC9AFD6AE74D9  2017-06-30      3.988    0.000002  0.007758   

                                            geometry  
0  POLYGON ((-13350814.27765598 4745388.128776118...  

[1 rows x 79 columns]

In [77]:
treatments.columns

Index([      u'SUID',        u'ORG', u'ACTIVITY_C',   u'ACTIVITY',
       u'LOCAL_QUAL', u'ASU_NBR_UN',    u'ASU_UOM', u'ADMIN_REGI',
       u'ADMIN_FORE', u'ADMIN_DIST', u'STATE_ABBR', u'OWNERSHIP_',
       u'PROC_REGIO', u'PROC_FORES', u'LAND_SUITA', u'PRODUCTIVI',
            u'SLOPE',  u'ELEVATION',     u'ASPECT', u'MGT_AREA_C',
       u'MGT_PRESCR', u'NBR_UNITS_', u'NBR_UNITS1',        u'UOM',
       u'DATE_PLANN', u'DATE_ACCOM', u'DATE_COMPL', u'FISCAL_YEA',
       u'FISCAL_Y_1', u'FISCAL_Y_2', u'FY_PLANNED',   u'KEYPOINT',
        u'FUND_CODE', u'METHOD_COD', u'EQUIPMENT_', u'COST_PER_U',
       u'NEPA_PROJE', u'NEPA_DOC_N', u'IMPLEMENTA', u'IMPLEMEN_1',
       u'IMPLEMEN_2', u'ACCOMPLISH', u'ACCOMPLI_1', u'ACTIVITY_1',
       u'ACTIVITY_U',   u'FACTS_ID',    u'SUBUNIT', u'FEATURE_TY',
       u'TREATMENT_', u'ACTIVITY_2', u'ACTIVITY_S', u'WORKFORCE_',
       u'NEPA_PRO_1',    u'SUID_CN',      u'ISWUI',       u'CWPP',
           u'CAT_NM',  u'EQUIPMENT',     u'METHOD', u'TREATMEN

In [78]:
treatments.REV_DATE.unique()

array(['2015-12-09', None, '2010-12-20', '2009-04-16', '2009-10-16',
       '2014-07-17', '2011-01-19', '2009-04-20', '2008-10-30',
       '2009-10-08', '2011-10-04', '2010-02-05', '2011-10-03',
       '2011-01-27', '2012-01-26', '2009-03-05', '2011-10-21',
       '2014-08-27', '2008-11-03', '2009-11-02', '2012-01-31',
       '2012-01-13', '2016-01-25', '2012-10-03', '2012-03-05',
       '2008-07-22', '2009-10-21', '2009-09-10', '2010-12-16',
       '2009-10-05', '2014-04-01', '2013-09-30', '2014-08-26',
       '2009-10-14', '2009-10-15', '2011-09-20', '1999-11-30',
       '2009-05-26', '2015-02-24', '2015-10-13', '2015-05-14',
       '2008-11-13', '2014-03-10', '2011-09-29', '2012-10-11',
       '2009-07-17', '2016-03-28', '2009-03-24', '2015-04-15',
       '2008-07-14', '2009-09-06', '2009-05-20', '2012-08-02',
       '2009-05-21', '2008-11-12', '2009-07-21', '2015-07-06',
       '2016-11-16', '2015-09-18', '2010-07-01', '2015-04-22',
       '2014-09-30', '2008-11-24', '2015-07-14', 

In [84]:
#treatments['tdate'] = treatments.REV_DATE.str.split('-')[0]
treatments['date'] = treatments.REV_DATE.str.split('-')

In [85]:
treatments['date'].sort_values().head()

4142    [1999, 11, 30]
733     [1999, 11, 30]
734     [1999, 11, 30]
735     [1999, 11, 30]
736     [1999, 11, 30]
Name: date, dtype: object

In [92]:
treatments['date'].dropna(inplace=True)

In [110]:
treatments['date'] = treatments['date'].apply(lambda x: x[0])

In [111]:
treatments.head()

SUID     ORG ACTIVITY_C                          ACTIVITY  \
0  0519540030061003000  051954       1153  Piling of Fuels, Hand or Machine   
1  0519540030061003000  051954       4521                Precommercial Thin   
2  0519540110060006000  051954       4521                Precommercial Thin   
3  0519540110060006000  051954       1153  Piling of Fuels, Hand or Machine   
4  0519540110061006000  051954       1153  Piling of Fuels, Hand or Machine   

  LOCAL_QUAL  ASU_NBR_UN ASU_UOM ADMIN_REGI ADMIN_FORE ADMIN_DIST  ...   \
0        N/A         4.0   ACRES         05         19         54  ...    
1        N/A         4.0   ACRES         05         19         54  ...    
2        N/A         9.3   ACRES         05         19         54  ...    
3        N/A         9.3   ACRES         05         19         54  ...    
4        N/A        55.0   ACRES         05         19         54  ...    

   EDW_INSERT                             UK_HAZ           CRC_HAZ  \
0  2017-01-14  5215581010602_0519540030061003000  B1AFC9AFD6AE74D9   
1  2017-01-14  5215561010602_0519540030061003000  B1AFC9AFA68C5897   
2  2017-01-14  5215669010602_0519540110060006000  AB64CB1D8A447A02   
3  2017-01-14  5215699010602_0519540110060006000  AB64CB1DB18D5FC4   
4  2017-01-14  5215037010602_0519540110061006000  7BC02E64C2BED043   

   ETL_MODI_1 GIS_ACRES SHAPE_AREA  SHAPE_LEN  \
0  2017-06-30     3.988   0.000002   0.007758   
1  2017-06-30     3.988   0.000002   0.007758   
2  2017-06-30     9.299   0.000004   0.011711   
3  2017-06-30     9.299   0.000004   0.011711   
4  2017-06-30    55.395   0.000023   0.035460   

                                            geometry       date_list  date  
0  POLYGON ((-13350814.27765598 4745388.128776118...  [2015, 12, 09]  2015  
1  POLYGON ((-13350814.27765598 4745388.128776118...  [2015, 12, 09]  2015  
2  (POLYGON ((-13350510.36352649 4745385.08375463...  [2015, 12, 09]  2015  
3  (POLYGON ((-13350510.36352649 4745385.08375463...  [2015, 12, 09]  2015  
4  (POLYGON ((-13350498.54891169 4745384.95773907...  [2015, 12, 09]  2015  

[5 rows x 81 columns]

In [115]:
treatments.pivot_table(index='ACTIVITY', 
               columns='date',
               aggfunc='size',
               fill_value=0)

date                                                1999  2008  2009  2010  \
ACTIVITY                                                                     
Broadcast Burning - Covers a majority of the unit      0     0     3     2   
Burning of Piled Material                              3   172   230   107   
Chipping of Fuels                                      0   207   128    88   
Commercial Thin                                       19     1    22     5   
Compacting/Crushing of Fuels                           7     1    14     2   
Improvement Cut                                        0     0    10    27   
Piling of Fuels, Hand or Machine                       4   166   221    99   
Precommercial Thin                                    19   294   335   145   
Rearrangement of Fuels                                 0    31     9     5   
Salvage Cut (intermediate treatment, not regene...     0     0     1     0   
Sanitation Cut                                         0    99    42    49   
Thinning for Hazardous Fuels Reduction                 1    21    14    17   
Underburn - Low Intensity (Majority of Unit)           0     0     3     2   
Yarding - Removal of Fuels by Carrying or Dragging    12     5     6     2   

date                                                2011  2012  2013  2014  \
ACTIVITY                                                                     
Broadcast Burning - Covers a majority of the unit      3     0     0     1   
Burning of Piled Material                             57    34     5   105   
Chipping of Fuels                                      2    40     1    70   
Commercial Thin                                        8    42     1     0   
Compacting/Crushing of Fuels                           6    26     0     0   
Improvement Cut                                       14    13     0     1   
Piling of Fuels, Hand or Machine                      44    57    17   148   
Precommercial Thin                                    23   121    18    27   
Rearrangement of Fuels                                 9    13     0     0   
Salvage Cut (intermediate treatment, not regene...     0     0     0     0   
Sanitation Cut                                         0    33     0    14   
Thinning for Hazardous Fuels Reduction                 0     3     0   186   
Underburn - Low Intensity (Majority of Unit)           1     0     0     0   
Yarding - Removal of Fuels by Carrying or Dragging    16     2     0     0   

date                                                2015  2016  
ACTIVITY                                                        
Broadcast Burning - Covers a majority of the unit      0     0  
Burning of Piled Material                              0     9  
Chipping of Fuels                                      2     0  
Commercial Thin                                        0     0  
Compacting/Crushing of Fuels                           1     0  
Improvement Cut                                        3     2  
Piling of Fuels, Hand or Machine                      32    30  
Precommercial Thin                                    33    18  
Rearrangement of Fuels                                 7     2  
Salvage Cut (intermediate treatment, not regene...     0     0  
Sanitation Cut                                         0     0  
Thinning for Hazardous Fuels Reduction                 1     9  
Underburn - Low Intensity (Majority of Unit)           0     0  
Yarding - Removal of Fuels by Carrying or Dragging     0     0

In [116]:
treatments.pivot_table(index='ACTIVITY', 
               columns='date',
               aggfunc='size',
               fill_value=0).sum()

date
1999      65
2008     997
2009    1038
2010     550
2011     183
2012     384
2013      42
2014     552
2015      79
2016      70
dtype: int64

In [118]:
treatments.pivot_table(index='ACTIVITY', 
               columns='date',
               aggfunc='size',
               fill_value=0).sum().median()

283.5

to do:<br>
take all Tahoe buildings<br>
buffer by defensible space<br>
compute years since last treated<br>
sort by years since last treated<br>
for each year in range(2020,2035)<br>
  allocate(top 284 oldest)

Impact Assessment<br>

create landfire scenarios knocking back risk to lowest in category for each treatment area upon treatment
linearly return towards highest risk FM after Fire Return Interval